docs/introduction/how-to-work-with-money.md

```diff
@@ -0,0 +1,428 @@
+
+The *currency ID* (`int`) must be provided as a parameter.
+
+### priceWithCurrencyByDomainId
+
+It works similarly to `priceWithCurrency`, but it uses the basic frontend currency of the provided domain.
+
+The *domain ID* (`int`) must be provided as a parameter.
+
+### moneyFormat
+
+Formats the amount of money as a decimal number without any currency symbol.
+
+Three optional parameters can be provided:
+- *number of decimal places* - `null` by default (meaning all), it will [round](#rounding) the value if necessary
+- *decimal point character* - `"."` by default
+- *separator of thousands* - `""` by default
+
+```twig
+{# the "money" variable contains Money::create('1234.5670') #}
+
+{{ money|moneyFormat }}                  {# renders "1234.5670" #}
+{{ money|moneyFormat(0) }}               {# renders "1235" #}
+{{ money|moneyFormat(2, ',') }}          {# renders "1234,57" #}
+{{ money|moneyFormat(null, '.', ',') }}  {# renders "1,234.5670" #}
+```
+
+## Money in Javascript
+
+`Money` implements the `JsonSerializable` interface, which means it can be serialized into JSON format with `json_encode` into an object with the following structure:
+
+```json
+{
+  "amount": "1234.5670"
+}
+```
+
+This format can be readily used in Javascript when JSON is rendered in Twig:
+
+```twig
+{# the "money" variable contains Money::create('1234.5670') #}
+
+<div id="my-money" data-money="{{ money|json_encode }}"/>
+```
+
+```js
+(function ($) {
+
+    $(document).ready(function () {
+        var money = $('#my-money').data('money');
+
+        // Displays "1234.5670"
+        alert(money.amount);
+    });
+
+})(jQuery);
+```
+
+See the native Javascript method [`.toFixed()`](https://www.w3schools.com/jsref/jsref_tofixed.asp) and the function [`parseFloat()`](https://www.w3schools.com/jsref/jsref_parsefloat.asp) to see how to convert between decimal strings and numbers.
+
+## Money in Doctrine
+
+In Shopsys Framework, there is a custom Doctrine type `money` which can be used in similar fashion as `decimal` type.
+The entity property value will be automatically hydrated to an instance of `Money` if it's configured to use the type:
+
+```php
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ */
+class MyEntity
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Money\Money|null
+     *
+     * @ORM\Column(type="money", precision=20, scale=6, nullable=true)
+     */
+    protected $price;
+
+    // ...
+}
+```
+
+### In Parameters
+
+When you want to use a `Money` instance as a parameter in DQL, use the `getAmount` method in your [repository class](/docs/introduction/basics-about-model-architecture.md#repository):
+
+```php
+use Shopsys\FrameworkBundle\Component\Money\Money;
+
+// ...
+
+$priceLimit = Money::create(1000);
+
+/** @var \Doctrine\ORM\QueryBuilder $builder */
+$builder->setParameter('priceLimit', $priceLimit->getAmount());
+```
+
+### In Function Results
+
```

Comment:
I suggest to split the text into 2 parts together with examples (quick suggestion, please review text and usages):

Doctrine hydrates values to an instance of `Money`.
```php
/** @var \Doctrine\ORM\EntityManager $em */
$result = $em->createQuery('SELECT o.totalPriceWithVat FROM ' . Order::class . ' o WHERE o.id = :id')
    ->setParameter('id', 1)
    ->getSingleResult();
// The result is automatically hydrated into Money
$money = $result['totalPriceWithVat'];
```

But once you're working with aggregate functions such as `MIN()`, `MAX()`, `SUM()`, etc., Doctrine have no idea about the type.
When fetching a monetary value from a function, you have to convert the fetched decimal string into a `Money` instance yourself using `Money::create()`:
```php
$result = $em->createQuery('SELECT AVG(o.totalPriceWithVat) AS averagePriceWithVat FROM ' . Order::class . ' o')
    ->getSingleResult();
// The result of a function is a decimal string (eg. '19590.772727272727') and it must be converted manually
$money = Money::create($result['averagePriceWithVat']);
```

Reply:
I've split it in 4880c8fcf